### PR TITLE
Fixed controller/exception examples

### PR DESCRIPTION
**Pull request by ionas**
* 3.x style plural table magic singletons
* 3.x style view variable setting
* Self sufficient exception (fully qualified namespace)
* Code snippets should now run by copy + paste

### CODE REVIEW

File: en/development/errors.rst [hunk 1]
```diff
@@ -204,17 +205,18 @@ For more details on HTTP 4xx error status codes see :rfc:`2616#section-10.4`.
...
 You can throw these exceptions from you controllers to indicate failure states,
 or HTTP errors. An example use of the HTTP exceptions could be rendering 404
 pages for items that have not been found::
 
-    public function view($id)
-    {
-        $post = $this->Post->findById($id);
-        if (!$post) {
-            throw new NotFoundException('Could not find that post');
+    public function view($id = null) {
```
Comment (markstory) [inline en/development/errors.rst]: `{` go on a separate line. PSR2 please

File: en/development/errors.rst [hunk 2]
```diff
@@ -204,17 +205,18 @@ For more details on HTTP 4xx error status codes see :rfc:`2616#section-10.4`.
...
+        $post = $this->Posts->findById($id)->first();
```
Comment (markstory) [inline en/development/errors.rst]: Perhaps we should use `get()` instead.

Comment (ionas) [inline en/development/errors.rst]: Yep. Exactly. The question is if to use RecordNotFoundException or NotFoundException.
Another NotFoundException example might be better?

Comment (ionas) [inline en/development/errors.rst]: Also, I don't like how the example appears twice on the page and was and is exactly the same.

File: en/development/errors.rst [hunk 3]
```diff
@@ -204,17 +205,19 @@ For more details on HTTP 4xx error status codes see :rfc:`2616#section-10.4`.
...
 
-    public function view($id)
+    public function view($id = null)
     {
-        $post = $this->Post->findById($id);
-        if (!$post) {
-            throw new NotFoundException('Could not find that post');
+        $post = $this->Posts->findById($id)->first();
+        if (empty($post)) {
+            throw new \Cake\Network\Exception\NotFoundException(__(' found'));
```
Comment (markstory) [inline en/development/errors.rst]: Missing a not here.

File: en/development/errors.rst [hunk 4]
```diff
@@ -204,17 +205,19 @@ For more details on HTTP 4xx error status codes see :rfc:`2616#section-10.4`.
...
 
-    public function view($id)
+    public function view($id = null)
     {
-        $post = $this->Post->findById($id);
-        if (!$post) {
-            throw new NotFoundException('Could not find that post');
+        $post = $this->Posts->findById($id)->first();
+        if (empty($post)) {
+            throw new \Cake\Network\Exception\NotFoundException(__('Post not found'));
```
Comment (antograssiot) [inline en/development/errors.rst]: in most places in the docs, we add a `use` statement instead of a FCN

Comment (antograssiot) [inline en/development/errors.rst]: @ionas ping. I can update post-merge if you prefer

File: en/development/errors.rst [hunk 5]
```diff
@@ -317,7 +320,7 @@ be thrown from a number of CakePHP core components:
 
 .. php:exception:: RecordNotFoundException
 
-   The requested record could not be found.
+   The requested record could not be found. This will also set http response headers to 404.
```
Comment (dereuromark) [inline en/development/errors.rst]: http should be uppercase in language. Can also do post nerge.

File: en/development/errors.rst (hunk 4 above)
Comment (ionas) [inline en/development/errors.rst]: This is so that you can copy and paste the snipplet. Should I put a use ontop instead?

Comment (antograssiot) [inline en/development/errors.rst]: yes please

File: en/development/errors.rst [hunk 6]
```diff
@@ -204,17 +205,21 @@ For more details on HTTP 4xx error status codes see :rfc:`2616#section-10.4`.
 
 For more details on HTTP 5xx error status codes see :rfc:`2616#section-10.5`.
 
+
 You can throw these exceptions from you controllers to indicate failure states,
```
Comment (antograssiot) [inline en/development/errors.rst]: @ions Just because I sde it now, would you please update also ``from you**r** controller`` here ?

**Comment by antograssiot** [comment]
looks good, I'll have a look in the context to see why we have twice the same code sample in the same page. We could probably do better here but it can come in a separate PR.

**Comment by ionas** [comment]
@antograssiot Yes, the next update I will do will be towards explaining how to use HttpException for other HTTP status code related errors where there are no exceptions build in, as soon as my other PR hits.

**Pull request merged**